server/services/v1/search_reader.go

```diff
@@ -89,11 +93,22 @@ func (p *page) readRow(row *Row) bool {
 			continue
 		}
 
-		// set the raw data now after marshaling it
-		if row.Data.RawData, p.err = jsoniter.Marshal(doc); p.err != nil {
+		// unmarshal the row data
+		var rawData []byte
+		if rawData, p.err = jsoniter.Marshal(doc); p.err != nil {
 			return false
 		}
-
+		// apply the field selection
+		if p.fieldSelection != nil {
+			newValue, err := p.fieldSelection.Apply(rawData)
+			if ulog.E(err) {
```

Comment:
What about user input is a list but in code, we can build the fields using `read.FieldFactory` then we can still use this method but user input is simple.

Reply:
I did make the change but doesn't look too intuitive to implicitly assume `fields` array to inclusion list. It may not appear intuitive if a user is building `[a, b]` for inclusion list and `[a: false, b: false]` for exclusion. Even if this array is easy to use, we should rather have `include` and `exclude` explicitly

Reply:
No, I was not saying it to have `[a: false, b: false]`. I was saying to just have an array `[a, b]`, similarly for exclude it will be an array.